File: Examples/ObjectiveC/MoPubRewardedAdapterExample/MoPubRewardedAdapterExample/ViewController.m
/*
 * Version for iOS © 2015–2021 YANDEX
 *
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at https://yandex.com/legal/mobileads_sdk_agreement/
 */

#import <mopub-ios-sdk/MoPub.h>
#import "ViewController.h"

// Replace XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX with Ad Unit ID generated at https://app.mopub.com
static NSString *const kMoPubBlockID = @"XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";

@interface ViewController () <MPRewardedAdsDelegate>

@property (weak, nonatomic) IBOutlet UIButton *loadButton;

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    MPMoPubConfiguration *configuration =
        [[MPMoPubConfiguration alloc] initWithAdUnitIdForAppInitialization:kMoPubBlockID];
    __typeof(self) __weak weakSelf = self;
    [[MoPub sharedInstance] initializeSdkWithConfiguration:configuration completion:^{
        dispatch_async(dispatch_get_main_queue(), ^{
            weakSelf.loadButton.userInteractionEnabled = YES;
        });
    }];
}

- (IBAction)loadAd
{
    [MPRewardedAds setDelegate:self forAdUnitId:kMoPubBlockID];
    [MPRewardedAds loadRewardedAdWithAdUnitID:kMoPubBlockID withMediationSettings:nil];
}

- (IBAction)presentAd
{
    MPReward *reward = [MPRewardedAds selectedRewardForAdUnitID:kMoPubBlockID];
    [MPRewardedAds presentRewardedAdForAdUnitID:kMoPubBlockID fromViewController:self withReward:reward];
}

#pragma mark - MPRewardedAdsDelegate

- (void)rewardedAdShouldRewardForAdUnitID:(NSString *)adUnitID reward:(MPReward *)reward
{
    NSString *message = [NSString stringWithFormat:@"Rewarded ad did reward: %@ %@", reward.amount, reward.currencyType];
    NSLog(@"%@", message);
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Reward"
                                                                             message:message
                                                                      preferredStyle:UIAlertControllerStyleAlert];
    [alertController addAction:[UIAlertAction actionWithTitle:@"Ok" style:UIAlertActionStyleCancel handler:nil]];
    [self.presentedViewController presentViewController:alertController animated:YES completion:nil];
}

- (void)rewardedAdDidLoadForAdUnitID:(NSString *)adUnitID
{
    NSLog(@"Rewarded ad loaded");
}

- (void)rewardedAdDidFailToLoadForAdUnitID:(NSString *)adUnitID error:(NSError *)error
{
    NSLog(@"Loading failed. Error: %@", error);
}

@end

File: Examples/Swift/MoPubRewardedAdapterExample/MoPubRewardedAdapterExample/ViewController.swift
/*
 * Version for iOS © 2015–2021 YANDEX
 *
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at https://yandex.com/legal/mobileads_sdk_agreement/
 */

import MoPubSDK

class ViewController: UIViewController {
    
    // Replace XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX with Ad Unit ID generated at https://app.mopub.com
    let adUnitID = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";

    @IBOutlet weak var loadButton: UIButton!

    override func viewDidLoad() {
        super.viewDidLoad()

        initializeMoPub()
    }

    @IBAction func loadAd(_ sender: UIButton) {
        MPRewardedAds.loadRewardedAd(withAdUnitID: adUnitID, withMediationSettings: [])
    }
    
    @IBAction func showAd(_ sender: UIButton) {
        let reward = MPRewardedAds.selectedReward(forAdUnitID: adUnitID)
        MPRewardedAds.presentRewardedAd(forAdUnitID: adUnitID, from: self, with: reward)
    }
    
    private func initializeMoPub() {
        MPRewardedAds.setDelegate(self, forAdUnitId: adUnitID)
        let configuration = MPMoPubConfiguration(adUnitIdForAppInitialization: adUnitID)
        MoPub.sharedInstance().initializeSdk(with: configuration){ [weak self] in
            DispatchQueue.main.async { [weak self] in
                self?.loadButton.isUserInteractionEnabled = true
            }
        }
    }
}

extension ViewController: MPRewardedAdsDelegate {

    func rewardedAdShouldReward(forAdUnitID adUnitID: String!, reward: MPReward!) {
        let amount = reward.amount!
        let currencyType = reward.currencyType!
        let message = "Rewarded ad did reward: \(amount) \(currencyType)"
        let allertController = UIAlertController(title: "Reward", message: message, preferredStyle: .alert)
        allertController.addAction(UIAlertAction(title: "Ok", style: .cancel, handler: nil))
        presentedViewController?.present(allertController, animated: true, completion: nil)
    }

    func rewardedAdDidLoad(forAdUnitID adUnitID: String!) {
        print("Rewarded ad did load")
    }

    func rewardedAdDidFailToLoad(forAdUnitID adUnitID: String!, error: Error!) {
        print("Rewarded ad did fail to load ad with error: \(error.localizedDescription)")
    }

    func rewardedAdDidFailToShow(forAdUnitID adUnitID: String!, error: Error!) {
        print("Rewarded ad did fail to present ad with error: \(error.localizedDescription)")
    }
}
